site/src/comparison.rs

```diff
@@ -423,6 +439,102 @@ impl ComparisonSummary {
     }
 }
 
+/// Writes a Markdown table containing summary of relevant results.
+pub fn write_summary_table(
+    primary: &ComparisonSummary,
+    secondary: &ComparisonSummary,
+    result: &mut String,
+) {
+    use std::fmt::Write;
+
+    fn render_stat<F: FnOnce() -> Option<f64>>(count: usize, calculate: F) -> String {
+        let value = if count > 0 { calculate() } else { None };
+        value
+            .map(|value| format!("{value:.1}%"))
+            .unwrap_or_else(|| "N/A".to_string())
+    }
+
+    writeln!(
+        result,
+        r#"| | Regressions 😿 <br />(primary) | Regressions 😿 <br />(secondary) | Improvements 🎉 <br />(primary) | Improvements 🎉 <br />(secondary) | All 😿 🎉 <br />(primary) |
+|:---:|:---:|:---:|:---:|:---:|:---:|"#
+    )
+    .unwrap();
+    writeln!(
+        result,
+        "| count[^1] | {} | {} | {} | {} | {} |",
+        primary.num_regressions,
+        secondary.num_regressions,
+        primary.num_improvements,
+        secondary.num_improvements,
+        primary.num_regressions + primary.num_improvements
+    )
+    .unwrap();
+
+    writeln!(
+        result,
+        "| mean[^2] | {} | {} | {} | {} | {:.1}% |",
+        render_stat(primary.num_regressions, || Some(
+            primary.arithmetic_mean_of_regressions()
+        )),
+        render_stat(secondary.num_regressions, || Some(
+            secondary.arithmetic_mean_of_regressions()
+        )),
+        render_stat(primary.num_improvements, || Some(
+            primary.arithmetic_mean_of_improvements()
+        )),
+        render_stat(secondary.num_improvements, || Some(
+            secondary.arithmetic_mean_of_improvements()
+        )),
+        primary.arithmetic_mean_of_changes()
+    )
+    .unwrap();
+
+    let largest_change = primary
+        .most_relevant_changes()
+        .iter()
+        .fold(0.0, |accum: f64, item| {
+            let change = item.map(|v| v.relative_change() * 100.0).unwrap_or(0.0);
+            accum.max(change)
+        });
+
+    writeln!(
+        result,
+        "| max | {} | {} | {} | {} | {:.1}% |",
+        render_stat(primary.num_regressions, || primary
+            .largest_regression()
+            .map(|r| r.relative_change() * 100.0)),
+        render_stat(secondary.num_regressions, || secondary
+            .largest_regression()
+            .map(|r| r.relative_change() * 100.0)),
+        render_stat(primary.num_improvements, || primary
+            .largest_improvement()
+            .map(|r| r.relative_change() * 100.0)),
+        render_stat(secondary.num_improvements, || secondary
+            .largest_improvement()
+            .map(|r| r.relative_change() * 100.0)),
+        largest_change
+    )
+    .unwrap();
+
+    if !primary.errors_in.is_empty() {
+        write!(
+            result,
+            "- Benchmark(s) {} started failing to build",
```

Comment:
I think it might be better do the following as I think it might be hard to read the way it's phrased now:

The following benchmarks failed to build:
- serde-1.0.135
- syn-1.0.89

Reply:
Good point, I changed it.